spec/factories/applicants.rb

```diff
@@ -23,5 +27,14 @@
         )
       end
     end
+
+    after(:create) do |applicant, evaluator|
+      if evaluator.num_bank_accounts > 0
+        provider = create :bank_provider, applicant: applicant
+        evaluator.num_bank_accounts.times do
+          create :bank_account, bank_provider: provider
+        end
+      end
+    end
```

Comment:
I'd suggest that `with_bank_accounts` would better match to the exiting traits. Also I don't think it is obvious what this code does or how to use it, so a comment would be useful. Something like:

 _"add `with_bank_accounts: 2` to add two bank accounts to the applicant"_.

Also as the `transient` is only used within this `after` block I would move the code so that the `transient` declaration was immediately before the `after` block.

Reply:
`with_bank_accounts` is not really a trait, but agree it makes sense to keep the naming convention.